app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,193 @@
+package com.example.drawingactivity
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.drawingactivity.DialogUtils.InputLimitations.CHAR_LIMIT
+import com.example.drawingactivity.DialogUtils.InputLimitations.LAYER_COUNT_START
+import com.example.drawingactivity.DialogUtils.InputLimitations.NAME_EMPTY
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogUtils {
+
+    private var hintIndex: Int = LAYER_COUNT_START
```

Comment:
Unless you _really_ know what you're doing, introducing a static variable like this is dicey. And with what we talked about before, you don't need a hint index right? 🤔